app/halgo/Halgo/CLA.hs

```diff
@@ -23,12 +23,12 @@ import Halgo.CLA.Command.Transaction (transactionOpts)
 import Halgo.CLA.Type as Type
 
 globalOpts :: Parser GlobalOptions
-globalOpts = GlobalOptions <$> strOption (mconcat
+globalOpts = GlobalOptions <$> option (eitherReader networkParser) (mconcat
   [ long "network"
   , metavar "GENESIS-ID"
   , help "Genesis ID of the Algorand network to work in"
-  , value "testnet-v1.0"
-  , showDefaultWith T.unpack
+  , value TestnetV1
+  , showDefaultWith show
```

Comment:
I'm not sure, but probably `showDefaultWith show` is just `showDefault`.

Reply:
I'll check it